src/librustdoc/clean/mod.rs

```diff
@@ -2320,22 +2320,29 @@ impl Clean<Item> for (&hir::MacroDef<'_>, Option<Ident>) {
         let tts = item.ast.body.inner_tokens().trees().collect::<Vec<_>>();
         // Extract the spans of all matchers. They represent the "interface" of the macro.
         let matchers = tts.chunks(4).map(|arm| arm[0].span()).collect::<Vec<_>>();
+        let source = if item.ast.macro_rules {
+            format!(
+                "macro_rules! {} {{\n{}}}",
+                name,
+                matchers
+                    .iter()
+                    .map(|span| { format!("    {} => {{ ... }};\n", span.to_src(cx)) })
+                    .collect::<String>(),
+            )
+        } else {
+            // This code currently assumes that there will only be one or zero matchers, as syntax
+            // for multiple is not currently defined.
+            format!(
+                "pub macro {}({}) {{\n\t...\n}}",
```

Comment:
Should this also check the visibility in case `--document-private-items` was passed? You can do that with `cx.tcx.visibility(def_id)` (which I guess means there was no point to using `from_hir_id`, oops).

Reply:
There appears to be a 'vis' item on MacroDef, would that work, or do we need to get the visibility from the context?

Reply:
Sure, `vis` works just as well :)